indico/modules/events/registration/fields/sessions.py

```diff
@@ -0,0 +1,55 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2024 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from marshmallow import ValidationError, fields, validate
+
+from indico.core.db.sqlalchemy import db
+from indico.modules.events.registration.fields.base import RegistrationFormFieldBase
+from indico.modules.events.registration.models.registrations import RegistrationData
+from indico.modules.events.sessions.models.blocks import SessionBlock
+from indico.util.i18n import _
+
+
+class SessionsField(RegistrationFormFieldBase):
```

Comment:
Just a thought, what will happen when you clone an event that uses this field and also clone registrations? I think the cloned registrations will keep referencing IDs from the old event.

I see a few options here:

1) Do not allow cloning registrations when there's data for a session field. This is probably rather annoying
2) Never clone data for session fields
3) Depending on whether sessions are also being cloned (you have access to this information in the cloners), either apply the ID mapping (between session ids from the old and new event), or clone the field with empty data

Reply:
I have checked it and you're right, it copies the data from the previous session blocks.
I think the best solution is ID mapping, but it's not trivial since the registration cloner has no access to the sessions mapper.

I think it may be good to solve it using a signal after cloning, but I didn't try yet. Do you have any better idea?

Reply:
Add the session cloner to the `uses` set of the RegistrationCloner:

See the relevant docstring for it here:

```
    #: cloners that must run before this one (if enabled), but this
    #: one runs even if they are not enabled
    uses = ...
```

With `uses = {'sessions'}` you will have access to the `session_block_map` from that cloner.